components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js

```diff
@@ -431,3 +431,76 @@ function viewFence(geoFenceElement,id) {
+            leafletLatLngs.push({lat: pItem[1], lng: pItem[0]});
+        });
+        geometryShape = new L.Polygon(leafletLatLngs);
+        map.addLayer(geometryShape);
+    }
+
+    var geoPublicUri = $("#geo-charts").data("geo-public-uri");
+
+    if(id=="Stationery"){
+
+        $('#templateLoader').load(geoPublicUri + "/assets/html_templates/view_fence_popup.html #viewStationeryAlert", function () {
+            var popupTemplate = $('#templateLoader').find('#viewStationeryAlert');
+            popupTemplate.find('#exportGeoJson').attr('leaflet_id', geometryShape._leaflet_id);
+            popupTemplate.find('#hideViewFence').attr('leaflet_id', geometryShape._leaflet_id);
+            popupTemplate.find('#viewAreaTime').html(stationeryTime);
+            geometryShape.bindPopup(popupTemplate.html(), {closeButton: true}).openPopup();
+            // transparent the layer .leaflet-popup-content-wrapper
+            $(geometryShape._popup._container.childNodes[0]).css("background", "rgba(255,255,255,0.8)");
+
+        });
+    } else if(id=="WithIn"){
+
+        $('#templateLoader').load(geoPublicUri + "/assets/html_templates/view_fence_popup.html #viewWithinAlert", function () {
+            var popupTemplate = $('#templateLoader').find('#viewWithinAlert');
+            popupTemplate.find('#exportGeoJson').attr('leaflet_id', geometryShape._leaflet_id);
+            popupTemplate.find('#hideViewFence').attr('leaflet_id', geometryShape._leaflet_id);
+            popupTemplate.find('#viewAreaName').html(areaName);
+            popupTemplate.find('#withinAlertForm').attr('area-name', areaName);
+            popupTemplate.find('#withinAlertForm').attr('query-name', queryName);
+            geometryShape.bindPopup(popupTemplate.html(), {closeButton: true}).openPopup();
+            // transparent the layer .leaflet-popup-content-wrapper
+            $(geometryShape._popup._container.childNodes[0]).css("background", "rgba(255,255,255,0.8)");
+        });
+    } else if(id=="Exit"){
+
+        $('#templateLoader').load(geoPublicUri + "/assets/html_templates/view_fence_popup.html #viewExitAlert", function () {
+            var popupTemplate = $('#templateLoader').find('#viewExitAlert');
+            popupTemplate.find('#exportGeoJson').attr('leaflet_id', geometryShape._leaflet_id);
+            popupTemplate.find('#hideViewFence').attr('leaflet_id', geometryShape._leaflet_id);
+            popupTemplate.find('#viewAreaName').html(areaName);
+            popupTemplate.find('#exitAlertForm').attr('area-name', areaName);
+            popupTemplate.find('#exitAlertForm').attr('query-name', queryName);
+            geometryShape.bindPopup(popupTemplate.html(), {closeButton: true}).openPopup();
+            // transparent the layer .leaflet-popup-content-wrapper
+            $(geometryShape._popup._container.childNodes[0]).css("background", "rgba(255,255,255,0.8)");
+        });
+    } else if(id=="Traffic"){
```

Comment:
@dilee is there any reason to have this empty if block? Also, I found some formatting issues in line 476, 489, 502. Can you check and fix these?

Reply:
Fixed in https://github.com/wso2/carbon-device-mgt/pull/950